docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,118 @@
+    * an **event** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional message text.
+  * **Alarms**:
+    * similar to Events, but the user or operator of the system has to take action to resolve the alarm
+    * an **alarm** is triggered by a **data point** (e.g. when it changes to a specific value as `0` or `1`), and contains an optional alarm text.
+
+## thin-edge plugins and capabilities
+
+**thin-edge** realizes high level cloud functionality (e.g. _configuration management_ or _log management_) with **plugins**.
+  * a **plugin** accesses _ressources_ and _services_ of the device, as e.g.
+      * the _package manager_ is accessed by _software managemennt_
+      * _configuration files_ are accessed by _configuration management_
+  * a **plugin** can be an (external) executable (e.g. as the `c8y_configuration_plugin` for _configuration management_)
+    or a thin-edge built-in software component (e.g. as for _software management_)
+  * usually a **plugin** runs on the **main-device**; thus it can access the _resources_ of the **main-device** directly
+  * to access _ressources_ of an **external child-device** a **plugin** needs another component, referred as **child-device agent**
+    * a **child-device agent** is the counterpart of a **plugin**, that takes the responsibility to access to the **external child-device's** _resources_
+    * a **child-device agent** can be installed and executed on the **external child-device**, or on the **main-device**
+      * if it runs in the **external child-device** it can access the _resources_ directly
+      * if it runs on the **main-device** it can use any (low-level) interfaces the **external child-device** provides to access those resources
+        * One main reason to install the **child-device agent** on the **main-device** is, when the **external child-device** cannot or shall not be altered.
+    * a **child-device agent** must interact with the **plugin**, based on the **protocol** defined by the **plugin**;
+      that could be e.g.:
+      * to listen and react to requests of the **plugin**, e.g. on MQTT
+      * to provide/consume files to/from the **plugin** on purpose, e.g. via HTTP
+
+A **plugin** can be described with a **capability**.
+   * a **capability** has a unique name, that references the **plugin** it describes (e.g. `tedge_config`)
+   * a **capability** implicitely claims the **protocol** the **plugin** defines
+   * whenever a **plugin** provides the same **capability** as a **child-device agent** supports, both are compatible and can interact with each other
+   * whenever one **plugin** provides the same **capability** as another one, it can replace the latter
+     * e.g. when a `c8y_configuration_plugin` and an `azure_configuration_plugin` provide the same **capability**, each can be replaced with the other one
```

Comment:
```suggestion
   * whenever a **plugin** provides the same **capability** as a **child-device agent** supports, both are compatible and can interact with each other
   * whenever one **plugin** provides the same **capability** as another one, it can replace the latter
     * e.g. when a `c8y_configuration_plugin` and an `azure_configuration_plugin` provide the same **capability**, each can be replaced with the other one
   * **capabilities** are especially used for provisioning of **external child-devices**, to align and connect all **child-device agent(s)** and the coresponding **plugins**
```